rct229/schema/validate.py

```diff
@@ -19,7 +19,111 @@
+    rmd
+
+    Returns list of mismatched fluid loop ids
+    -------
+
+    """
+    mismatch_list = []
+
+    fluid_loop_id_list = find_all("$.fluid_loops[*].id", rmd)
+    fluid_loop_id_list.extend(find_all("$.fluid_loops[*].child_loops[*].id", rmd))
+
+    referenced_id_list = find_all("$.chillers[*].cooling_loop", rmd)
+    referenced_id_list.extend(find_all("$.chillers[*].condensing_loop", rmd))
+    referenced_id_list.extend(find_all("$.chillers[*].heat_recovery_loop", rmd))
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].heating_system.hot_water_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].heating_system.water_source_heat_pump_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].cooling_system.chilled_water_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*].cooling_system.condenser_water_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].spaces[*].miscellaneous_equipment[*].energy_from_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].spaces[*].miscellaneous_equipment[*].remaining_fraction_to_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].terminals[*].cooling_from_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.buildings[*].building_segments[*].zones[*].terminals[*].heating_from_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.service_water_heating_equipment[*].hot_water_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.heat_rejections[*].loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.boilers[*].loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.service_water_heating_equipment[*].hot_water_loop",
+            rmd,
+        )
+    )
+    referenced_id_list.extend(
+        find_all(
+            "$.external_fluid_sources[*].loop",
+            rmd,
+        )
+    )
+
+    for fluid_loop_id in referenced_id_list:
+        if fluid_loop_id not in fluid_loop_id_list:
+            mismatch_list.append(fluid_loop_id)
```

Comment:
One thing I was thinking is whether we allow orphan fluid loops in the project.
I can see zone does not have this issue but not sure about fluid.
If that is the logic we want to add, then the HVAC association check will also need update.

Reply:
Need to run black before merging.

Reply:
A particular BEM tool might have errors arise from orphan model elements, but I don't think it will be universal. I don't think we want to add logic that checks if there are orphans.